test/e2e/tests/tokens/nft/auto-detect-nft.spec.ts

```diff
@@ -0,0 +1,49 @@
+import { withFixtures } from '../../../helpers';
+import FixtureBuilder from '../../../fixture-builder';
+import HeaderNavbar from '../../../page-objects/pages/header-navbar';
+import Homepage from '../../../page-objects/pages/homepage';
+import PrivacySettings from '../../../page-objects/pages/settings/privacy-settings';
+import SettingsPage from '../../../page-objects/pages/settings/settings-page';
+import { loginWithBalanceValidation } from '../../../page-objects/flows/login.flow';
+import { setupAutoDetectMocking } from './mocks';
+
+describe('NFT detection', function () {
+  /**
+   * TODO Revisit this test once we enable nft auto detection by default. Use .withPreferencesControllerNftDetectionEnabled().
+   */
+  it('displays NFT media', async function () {
+    const driverOptions = { mock: true };
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().withNetworkControllerOnMainnet().build(),
+        driverOptions,
+        title: this.test?.fullTitle(),
+        testSpecificMock: setupAutoDetectMocking,
+      },
+      async ({ driver }) => {
+        await loginWithBalanceValidation(driver);
+
+        // navigate to security & privacy settings and toggle on NFT autodetection
```

Comment:
@chloeYue, this is a suggestion: could we include comments in the test console.log? 

I feel that when analyzing any flaky test, it would be helpful especially when actions and pages are repeated like visiting the settings twice or if the test has lot of switching windows.

Including the test console.log would be beneficial. You don't have to modify this PR, but if you think it would help please do include them.

Reply:
Hi @hjetpoluru , yes, I totally agree that we currently have very poor logging and need to add more `console.log` statements for page interactions. However, I try to put these logs in the page class functions instead of the test bodies. This way, when people call these functions, the logs are automatically added. Please see the current log on this branch for autodetect NFT:
<img width="1201" alt="Screenshot 2024-11-12 at 23 10 10" src="https://github.com/user-attachments/assets/428bebe6-5218-468c-ba96-9f291c47f6ff">